src/Nethermind/Nethermind.JsonRpc/Modules/Subscribe/SyncingSubscription.cs

```diff
@@ -0,0 +1,115 @@
+//  Copyright (c) 2021 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System.Threading.Tasks;
+using Nethermind.Blockchain;
+using Nethermind.JsonRpc.Modules.Eth;
+using Nethermind.Logging;
+
+namespace Nethermind.JsonRpc.Modules.Subscribe
+{
+    public class SyncingSubscription : Subscription
+    {
+        private readonly IBlockTree _blockTree;
+        private readonly ILogger _logger;
+        private bool IsSyncing { get; set; }
+        private long BestSuggestedNumber { get; set; }
+        
+        public SyncingSubscription(IBlockTree blockTree, ILogManager logManager)
+        {
+            _blockTree = blockTree;
+            _logger = logManager.GetClassLogger();
+            
+            IsSyncing = CheckSyncing();
+            if(_logger.IsTrace) _logger.Trace($"Syncing subscription {Id}: Syncing status on start is {IsSyncing}");
+            
+            _blockTree.NewBestSuggestedBlock += OnConditionsChange;
+            if(_logger.IsTrace) _logger.Trace($"Syncing subscription {Id} will track NewBestSuggestedBlocks");
+            
+            _blockTree.NewHeadBlock += OnConditionsChange;
+            if(_logger.IsTrace) _logger.Trace($"Syncing subscription {Id} will track NewHeadBlocks");
+        }
+
+        private bool CheckSyncing()
+        {
+            BestSuggestedNumber = _blockTree.FindBestSuggestedHeader().Number;
+            return BestSuggestedNumber > _blockTree.Head.Number + 8;
+        }
```

Comment:
Can we make it BlockFinder Extension and reuse it in all places?

Reply:
CheckSyncing in block finder? does not sound like what BlockFinder should be responsible for

Reply:
No in extension method or somewhere else as I think we duplicate this check in 2 places now.